tests/spdx/validation/test_external_package_ref_validator.py

```diff
@@ -13,25 +13,147 @@
+def test_valid_external_package_ref(category, reference_type, locator):
+    external_package_ref = ExternalPackageRef(category, reference_type, locator, "externalPackageRef comment")
     validation_messages: List[ValidationMessage] = validate_external_package_ref(external_package_ref, "parent_id")
 
     assert validation_messages == []
 
 
-@pytest.mark.parametrize("external_package_ref, expected_message",
-                         [(external_package_ref_fixture(),
-                           "TBD"),
+@pytest.mark.parametrize("category, reference_type, locator, expected_message",
+                         [(
+                          ExternalPackageRefCategory.SECURITY, "cpe22Typo", "cpe:/o:canonical:ubuntu_linux:10.04:-:lts",
+                          "externalPackageRef type in category SECURITY must be one of [cpe22Type, cpe23Type, advisory, fix, url, swid], but is: cpe22Typo"),
+                          (ExternalPackageRefCategory.PACKAGE_MANAGER, "nugat",
+                           "cpe:/o:canonical:ubuntu_linux:10.04:-:lts",
+                           "externalPackageRef type in category PACKAGE_MANAGER must be one of [maven-central, npm, nuget, bower, purl], but is: nugat"),
+                          (ExternalPackageRefCategory.PERSISTENT_ID, "git-oid",
+                           "cpe:/o:canonical:ubuntu_linux:10.04:-:lts",
+                           "externalPackageRef type in category PERSISTENT_ID must be one of [swh, gitoid], but is: git-oid")
+                          ])
+def test_invalid_external_package_ref_types(category, reference_type, locator, expected_message):
+    external_package_ref = ExternalPackageRef(category, reference_type, locator, "externalPackageRef comment")
+    parent_id = "SPDXRef-Package"
+    validation_messages: List[ValidationMessage] = validate_external_package_ref(external_package_ref, parent_id)
+
+    expected = ValidationMessage(expected_message,
+                                 ValidationContext(parent_id=parent_id,
+                                                   element_type=SpdxElementType.EXTERNAL_PACKAGE_REF,
+                                                   full_element=external_package_ref))
+
+    assert validation_messages == [expected]
+
+
+CPE22TYPE_REGEX = r'^c[pP][eE]:/[AHOaho]?(:[A-Za-z0-9._\-~%]*){0,6}$'
+CPE23TYPE_REGEX = r'^cpe:2\.3:[aho\*\-](:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!"#$$%&\'\(\)\+,\/:;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\*\-]))(:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!"#$$%&\'\(\)\+,\/:;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){4}$'
+MAVEN_CENTRAL_REGEX = r'^[^:]+:[^:]+(:[^:]+)?$'
+NPM_REGEX = r'^[^@]+@[^@]+$'
+NUGET_REGEX = r'^[^/]+/[^/]+$'
+BOWER_REGEX = r'^[^#]+#[^#]+$'
+PURL_REGEX = r'^pkg:.+(\/.+)?\/.+(@.+)?(\?.+)?(#.+)?$'
+SWH_REGEX = r'^swh:1:(snp|rel|rev|dir|cnt):[0-9a-fA-F]{40}$'
+GITOID_REGEX = r'^gitoid:(blob|tree|commit|tag):(sha1:[0-9a-fA-F]{40}|sha256:[0-9a-fA-F]{64})$'
```

Comment:
I would import these regex statements from `external_package_ref_validator.py` to prevent possible mismatch in the future if one of the regex needs changes and simply to avoid duplications.

Reply:
done